app/controllers/provider_interface/decisions_controller.rb

```diff
@@ -1,9 +1,39 @@
 module ProviderInterface
   class DecisionsController < ProviderInterfaceController
     before_action :set_application_choice
+    before_action :confirm_application_is_in_decision_pending_state, only: %i[new create]
     before_action :requires_make_decisions_permission
 
+    def new
+      @wizard = OfferWizard.new(offer_store,
+                                offer_context_params(@application_choice.course_option).merge!(current_step: 'select_option'))
+      @wizard.save_state!
+    end
+
+    def create
+      @wizard = OfferWizard.new(offer_store, { decision: selected_decision })
+
+      if @wizard.valid_for_current_step?
+
+        @wizard.save_state!
+
+        if @wizard.decision == 'rejection'
+          @wizard.clear_state!
+
+          redirect_to provider_interface_reasons_for_rejection_initial_questions_path(@application_choice)
+        else
+          redirect_to [:new, :provider_interface, @application_choice, :offer, @wizard.next_step]
```

Comment:
It seems that sometimes we lose the `@wizard.next_step` and we get into an inconsistent state and I get the following error:
```
undefined method `new_provider_interface_application_choice_offer_url'
```
when it tries to build this path, maybe we can have a fallback if the next step is missing? this might be resolved by the clearing the wizard state above

Reply:
Can you point out a scenario when this is happenning? Is this after going back from the Rejections controller or are there any other scenarios as well?

Reply:
@cwrw So we attempted to amend the form object's `next_step` so it defaults to the one at the beginning of the flow. However, as the decisions page does not follow our path convention, that didn't work.

The temporary solution that we have put in place for now, so that the error can be handle gracefully, is on commit https://github.com/DFE-Digital/apply-for-teacher-training/pull/4218/commits/bf04a9b0174d012865705a61d4f886e21c66d18b. Do you have any thoughts on this? 

I think we can look more into it as part of change offer.

Reply:
I think that handling the error gracefully is much better, although still surprising to be redirected to the main new offer page, but happy to wait for the change offer work to tackle this later 👍